app/gui/src/project-view/components/DocumentationEditor/images.ts

```diff
@@ -0,0 +1,180 @@
+import { fetcherUrlTransformer } from '@/components/MarkdownEditor/imageUrlTransformer'
+import { Vec2 } from '@/util/data/vec2'
+import type { ToValue } from '@/util/reactivity'
+import { useToast } from '@/util/toast'
+import { computed, reactive, toValue } from 'vue'
+import type { Path } from 'ydoc-shared/languageServerTypes'
+import { Err, mapOk, Ok, Result, withContext } from 'ydoc-shared/util/data/result'
+import type { Uuid } from 'ydoc-shared/yjsModel'
+
+type UploadedImagePosition = { type: 'selection' } | { type: 'coords'; coords: Vec2 }
+
+/**
+ * A Project File management API for {@link useDocumentationImages} composable.
+ */
+interface ProjectFilesAPI {
+  projectRootId: Promise<Uuid | undefined>
+  readFileBinary(path: Path): Promise<Result<Blob>>
+  writeFileBinary(path: Path, content: Blob): Promise<Result>
+  pickUniqueName(path: Path, suggestedName: string): Promise<Result<string>>
+  ensureDirExists(path: Path): Promise<Result<void>>
+}
+
+const supportedImageTypes: Record<string, { extension: string }> = {
+  // List taken from https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types
+  'image/apng': { extension: 'apng' },
+  'image/avif': { extension: 'avif' },
+  'image/gif': { extension: 'gif' },
+  'image/jpeg': { extension: 'jpg' },
+  'image/png': { extension: 'png' },
+  'image/svg+xml': { extension: 'svg' },
+  'image/webp': { extension: 'webp' },
+  // Question: do we want to have BMP and ICO here?
```

Comment:
Talking with @Frizi we decided that we do not, but I forgot to remove this comment.